### DISCUSSION TITLE
Swagger docs shows query params as body request

### DISCUSSION THREAD

**Discussion by ecosciug** [fastapi/fastapi#9481] [Questions]
<img width="1383" alt="Снимок экрана 2023-05-01 в 11 34 20" src="https://user-images.githubusercontent.com/74615964/235435733-c4a0725f-a937-4e6c-b8cf-437801a1d8a1.png">
<img width="597" alt="Снимок экрана 2023-05-01 в 11 34 07" src="https://user-images.githubusercontent.com/74615964/235435736-c1fa70a5-a8f7-43f9-bdd8-3db3254c4912.png">
<img width="398" alt="Снимок экрана 2023-05-01 в 11 33 47" src="https://user-images.githubusercontent.com/74615964/235435738-b7314baa-f34a-4233-abc6-07569e0da0f7.png">

I have an endpoint, that takes as input from query params VehicleFilters, but transmission field is not shown in swagger as query param but as body request. I let as example param "field" and I want transmission to look like param field

_Originally posted by @ecosciug in https://github.com/tiangolo/fastapi/discussions/6748#discussioncomment-5770488_

**Comment by djimontyp** [comment]
@ecosciug 
You can use the arguments on the python class in the __init__ method. There are good examples in the documentation.
[Classes as Dependencies
](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/)

```from typing import Annotated

from fastapi import Depends, FastAPI

app = FastAPI()


fake_items_db = [{"item_name": "Foo"}, {"item_name": "Bar"}, {"item_name": "Baz"}]


class CommonQueryParams:
    def __init__(self, q: str | None = None, skip: int = 0, limit: int = 100):
        self.q = q
        self.skip = skip
        self.limit = limit


@app.get("/items/")
async def read_items(commons: Annotated[CommonQueryParams, Depends(CommonQueryParams)]):
    response = {}
    if commons.q:
        response.update({"q": commons.q})
    items = fake_items_db[commons.skip : commons.skip + commons.limit]
    response.update({"items": items})
    return response
```



I hope this helps you.

**Reply by ecosciug** [reply]
thanks for your reply, but this doesn't help, when you try to get as a query parameter a List of a enum it will display that field as body request not as query param.

**Comment by HeeebsInc** [comment]
@ecosciug have you found a solution for this?  I am facing a similar issue where the swagger UI is presenting a param as a request body.  This is for a get request, so it fails since there cannot be a body for these types of requests.

**Comment by JavierSanchezCastro** [comment]
Try that:

```python3
from fastapi import Depends, FastAPI, Query
from typing import Annotated
from pydantic import BaseModel, Field
from enum import Enum

app = FastAPI()

class MyClassEnum(str, Enum):
    D = "d"
    E = "e"
    
class MyClass(BaseModel):
    a: str = Query(default="abc")
    b: list[int] = Field(Query([]))
    c: list[MyClassEnum] = Field(Query([]))

@app.get("/")
def test(filters: Annotated[MyClass, Depends()]):
    return None
```
From https://stackoverflow.com/a/74124759/10763551 and https://github.com/tiangolo/fastapi/issues/4445#issuecomment-1117632409
